app/src/main/java/com/github/gogetters/letsgo/activities/MapsActivity.kt

```diff
@@ -74,32 +78,31 @@ class MapsActivity : AppCompatActivity(), OnMapReadyCallback, GoogleMap.OnMyLoca
 
         // Move the camera to EPFL
         mMap.moveCamera(CameraUpdateFactory.newLatLngZoom(EPFL, INIT_ZOOM))
+        mMap.setOnMarkerClickListener { marker ->
+            displayUserInfo(marker.tag as String)
+            true
+        }
 
         googleMap.setOnMyLocationButtonClickListener(this)
         googleMap.setOnMyLocationClickListener(this)
         enableLocation()
         sendLocation()
     }
 
-    /**
-     * Allows to set a locationSharingService that is used to share and retrieve locations of users
-     */
-    fun setLocationSharingService(locationSharingService: LocationSharingService) {
-        this.locationSharingService = locationSharingService
-    }
-
 
     private fun enableLocation() {
         if (!::mMap.isInitialized)
             return
 
         if (ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION)
-                == PackageManager.PERMISSION_GRANTED) {
+            == PackageManager.PERMISSION_GRANTED
```

Comment:
All of this could be packed into a utility function, since I have the same code in bluetooth as well..